### PR TITLE
Fix multi_choice

### PR DESCRIPTION
**Pull request by nop33**

### CODE REVIEW

File: indico/modules/events/registration/fields/choices_test.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            {'id': _id(1), 'places_limit': 0, 'is_billable': False, 'price': 0},
+            {'id': _id(2), 'places_limit': 0, 'is_billable': False, 'price': 0},
+            {'id': _id(3), 'places_limit': 0, 'is_billable': True, 'price': 10}
+        ]
+    }
+    return field
+
+
+def _update_data(data, changes):
+    data = dict(deepcopy(data))
```
Comment (mic4ael) [inline indico/modules/events/registration/fields/choices_test.py]: Why are you using `dict` here? Looking at the code I think you are always passing a dict so `deepcopy` will return an object of the same type.

Comment (ThiefMaster) [inline indico/modules/events/registration/fields/choices_test.py]: I think I'm sometimes passing `RegistrationFormItem.versioned_data` which is an `ImmutableMultiDict`

File: indico/modules/events/registration/fields/simple.py [hunk 2]
```diff
@@ -57,6 +64,9 @@ def get_field_merged_options(field, registration_data):
             if missing_option:
                 result['choices'].append(missing_option)
                 result['deletedChoice'].append(missing_option['id'])
+        else:
+            result['choices'].remove(_get_choice_by_id(val, result['choices']))
```
Comment (ThiefMaster) [inline indico/modules/events/registration/fields/simple.py]: This doesn't preserve the position, does it? (probably not a big deal though)

Comment (nop33) [inline indico/modules/events/registration/fields/simple.py]: No, it doesn't, but I also believe that it doesn't matter much.

**Comment by ThiefMaster** [comment]
Merged ebf1138

**Pull request closed without merging**